lib/home_screen/view/home_screen.dart

```diff
@@ -547,20 +554,23 @@ class _MyHomePageState extends State<MyHomePage> {
           ),
         ),
       ),
-      bottomNavigationBar: const BottomNavBar(),
+      bottomNavigationBar: BottomNavBar(pageIndex: 0, onChanged: onChanged),
```

Comment:
Instead of passing 0, pass pageIndex to pageIndex, else your bottomNavBar will not reflect the changes from navigating to a new page